src/qiskit_qec/decoders/hdrg_decoders.py

```diff
@@ -522,11 +524,251 @@ def peeling(self, erasure: PyGraph) -> List[int]:
             pendant_vertex = endpoints[0] if not tree.vertices[endpoints[0]] else endpoints[1]
             tree_vertex = endpoints[0] if pendant_vertex == endpoints[1] else endpoints[1]
             tree.vertices[tree_vertex].remove(edge)
-            if erasure[pendant_vertex]["syndrome"] and not erasure[pendant_vertex]["is_boundary"]:
+            if (
+                erasure[pendant_vertex].properties["syndrome"]
+                and not erasure[pendant_vertex].is_boundary
+            ):
                 edges.add(edge)
-                erasure[tree_vertex]["syndrome"] = not erasure[tree_vertex]["syndrome"]
-                erasure[pendant_vertex]["syndrome"] = False
+                erasure[tree_vertex].properties["syndrome"] = not erasure[tree_vertex].properties[
+                    "syndrome"
+                ]
+                erasure[pendant_vertex].properties["syndrome"] = False
+
+        return [erasure.edges()[edge].qubits[0] for edge in edges if erasure.edges()[edge].qubits]
+
+
+@dataclass
+class Cluster:
+    """
+    Cluster class for the ClAYG decoder.
+    FIXME: Remove, when ClAYG decoder uses Union Find infrastructure.
+    """
+
+    boundary: List[Tuple[int, int]]  # List[(edge, neighbour)]
+    fully_grown_edges: List[int]  # List[edge]
+    nodes: List[int]  # List[node_index]
+    atypical_nodes: List[int]  # List[node_index]
+
+
+class ClAYGDecoder(UnionFindDecoder):
```

Comment:
Maybe it's better for ClAYG to have its own PR?

Reply:
I accidentally created the custom-type-branch on top of the clayg branch, so it has some changes on it which depend on the ClAYG. I can try to do a rebase onto main, but it will probably be a mess.